app/components/autocomplete/at_mention/at_mention.js

```diff
@@ -50,74 +42,81 @@ export default class AtMention extends Component {
     constructor(props) {
         super(props);
 
-        const ds = new ListView.DataSource({
-            sectionHeaderHasChanged: (s1, s2) => s1 !== s2,
-            rowHasChanged: (r1, r2) => r1 !== r2
-        });
-        const data = {};
         this.state = {
-            active: false,
-            dataSource: ds.cloneWithRowsAndSections(data)
+            sections: []
         };
     }
 
     componentWillReceiveProps(nextProps) {
-        const {isSearch} = nextProps;
-        const regex = isSearch ? FROM_REGEX : AT_MENTION_REGEX;
-        const match = nextProps.postDraft.substring(0, nextProps.cursorPosition).match(regex);
-
-        if (!match || this.state.mentionComplete) {
+        const {inChannel, outChannel, teamMembers, isSearch, matchTerm, requestStatus} = nextProps;
+        if ((matchTerm !== this.props.matchTerm && matchTerm === null) || this.state.mentionComplete) {
+            // if the term changes but is null or the mention has been completed we render this component as null
             this.setState({
-                active: false,
-                matchTerm: null,
-                mentionComplete: false
+                mentionComplete: false,
+                sections: []
             });
             return;
+        } else if (matchTerm === null) {
+            // if the terms did not change but is null then we don't need to do anything
+            return;
         }
 
-        const matchTerm = isSearch ? match[1] : match[2];
-        if (matchTerm !== this.state.matchTerm) {
-            this.setState({
-                matchTerm
-            });
-
+        if (matchTerm !== this.props.matchTerm && requestStatus !== RequestStatus.STARTED) {
```

Comment:
I think checking the requestStatus here will prevent the autocomplete from searching past the first letter on a slower connection. Like they'll type "b" and then it'll search for users starting with "b", but then if they type "bob", it won't get updated results for "bob"

Reply:
Make sense I'll be changing this one